test/Scrutor.Tests/ScanningTests.cs

```diff
@@ -140,7 +175,7 @@ public void AutoRegisterAsMatchingInterface()
                     .AsMatchingInterface()
                     .WithTransientLifetime());
 
-            Assert.Equal(2, Collection.Count);
+            Assert.Equal(3, Collection.Count);

```

Comment:
I'm not really sure what the best direction is here. People might expect different things:
- Register as self
- Register as all implemented interfaces
- Register as the first interface
- Register as the last interface
- Register as some interface between :stuck_out_tongue_closed_eyes:
- All of the above

It's really hard to come up with a magic convention like this. The real solution, IMO, is to register as all of the above. I don't think it matters much. It's cheap to register services. If you care about the number of registered services, but still wanted something like this, you'd be better off with the fluent API. That lets you do all kinds of crazy stuff when it comes to conventions.